ipc/src/server.rs

```diff
@@ -172,7 +172,9 @@ impl<M: Metadata, S: Middleware<M>> ServerBuilder<M, S> {
 					})
 				})
 				.filter_map(|x| x)
-				.select(receiver.map_err(|e| {
+				// we use `select_with_weak` here, instead of `select`, to close the stream
+				// as soon as the ipc pipe is closed
+				.select_with_weak(receiver.map_err(|e| {
```

Comment:
maintains the stream as long as the connection exists. the outgoing stream may be closed earlier

Reply:
I don't get why we use `select_with_weak`, why not simlpy spawn the two streams separately? The point of `select` is to end the stream if any of the stream ends, now we have `select_with_weak` which "Finishes when strong stream finishes", so what's the point of using `select` at all if you can spawn the stream separately?